baseplate/lib/prometheus_metrics.py

```diff
@@ -88,3 +89,106 @@ def get_requests_total_metric(self) -> Counter:
     def get_active_requests_metric(self) -> Gauge:
         """Return the active_requests metrics"""
         return thrift_server_active_requests
+
+
+# http server labels and metrics
+http_server_histogram_labels = [
+    "http_method",
+    "http_endpoint",
+    "http_success",
+]
+http_server_requests_total_labels = [
+    "http_method",
+    "http_endpoint",
+    "http_success",
+    "http_response_code",
+]
+http_server_active_requests_labels = [
+    "http_method",
+    "http_endpoint",
+]
+
+http_server_latency_seconds = Histogram(
+    "http_server_latency_seconds",
+    "Description of histogram",
+    http_server_histogram_labels,
+    buckets=default_buckets,
+)
+http_server_request_size_bytes = Histogram(
+    "http_server_request_size_bytes",
+    "Description of histogram",
+    http_server_histogram_labels,
+    buckets=default_buckets,
+)
+http_server_response_size_bytes = Histogram(
+    "http_server_response_size_bytes",
+    "Description of histogram",
+    http_server_histogram_labels,
+    buckets=default_buckets,
+)
+http_server_requests_total = Counter(
+    "http_server_requests_total",
+    "Description of counter",
+    http_server_requests_total_labels,
+)
+http_server_active_requests = Gauge(
+    "http_server_active_requests",
+    "Description of gauge",
+    http_server_active_requests_labels,
+)
+
+
+class PrometheusHTTPServerMetrics:
+    def __init__(self) -> None:
+        pass
+
+    def latency_seconds_metric(self, tags: Dict[str, str]) -> Any:
```

Comment:
In what context will these functions be used?

Reply:
They're called from `observers/prometheus.py` in `on_finish`. It's not in this PR because it was already merged with the Thrift implementation. 

If we end up keeping this abstraction, we should in the future make it clear that this is implementing an interface.

Reply:
Can you provide a code sample? This naming just seems really odd to me and I'm curious if we can come up with a better name.